src/components/PopupWithForm.js

```diff
@@ -16,10 +18,19 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  setLoading(isLoading, savingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = savingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
+      this.setLoading(true);
       this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
```

Comment:
You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
When I add a then block in here, I now  get an error when I try to submit/delete a card, how do I resolve this issue